### ISSUE TITLE
`render` in controller should respect layout

### ISSUE THREAD

**Issue by fsateler** [github/view_component#391]
## Feature request

When using `render MyComponent.new` in a controller, the layout should be respected. See also https://github.com/github/view_component/issues/38#issue-487741087, which explicitly set the layout option.

### Motivation

I would like to stop using ActionView templates altogether. It is easier to work with a single view library than multiple ones, especially as "you need to switch from ActionView to ViewComponent" tends to be annoying when adding a small thing. Therefore, I'd like to have Page Components (eg, Users::ShowComponent, etc), and render them from the controller without going into an ActionView file first.

### More information

A simple solution like the one posted on https://github.com/github/view_component/issues/38#issue-487741087 is possible, but unfortunately uses a different view_context than for the layout itself. Therefore, sharing `content_for` with the layout is not possible.

**Comment by joelhawksley** [comment]
@fsateler I agree that this should be possible. Might you be willing to open a PR with failing tests for the APIs you think should work?

**Comment by mollerhoj** [comment]
Working on a relatively small project, this would be my exact use case as well. Would you consider allowing for a different name than "Component". It seems like a strange name of objects that represent entire views - Something along the lines of ViewModels seem like a better fit. Perhaps even just allowing placing the objects in the `view/` folder, so that the structure becomes

```
app/views/users/show.html.erb
app/views/users/show.rb    #replacing Users::ShowComponent with Users::Show
```

**Comment by joelhawksley** [comment]
@mollerhoj I hear you! 

We started with building ViewComponent to leverage the `views` folder, but it became pretty messy to make things work cleanly with Rails. I can see why it would make sense to include them in the `views` folder though, so maybe that's a Rails feature that could be added?

**Comment by mollerhoj** [comment]
I've created a new Issue to avoid hijacking this issue.

**Comment by fsateler** [comment]
This issue will be worked in upstream rails: https://github.com/rails/rails/pull/39869

**Comment by gap777** [comment]
@joelhawksley Is this issue really closed/resolved? In Rails 7.2.2.1, `render MyViewComponent.new(...)` seems to NOT include layout. 
I could use the workaround mentioned in https://github.com/ViewComponent/view_component/issues/38#issuecomment-1162287527, but really.... that shouldn't be necessary.
Thoughts?
